libs/@local/harpc/service/src/session/server/write.rs

```diff
@@ -0,0 +1,263 @@
+use bytes::{Buf, Bytes};
+use bytes_utils::SegmentedBuf;
+use harpc_wire_protocol::{
+    flags::BitFlagsOp,
+    payload::Payload,
+    protocol::{Protocol, ProtocolVersion},
+    request::id::RequestId,
+    response::{
+        begin::ResponseBegin,
+        body::ResponseBody,
+        flags::{ResponseFlag, ResponseFlags},
+        frame::ResponseFrame,
+        header::ResponseHeader,
+        kind::{ErrorCode, ResponseKind},
+        Response,
+    },
+};
+use tokio::sync::mpsc;
+
+pub(crate) struct ResponseWriter<'a> {
+    id: RequestId,
+    kind: ResponseKind,
+    index: usize,
+
+    buffer: SegmentedBuf<Bytes>,
+
+    tx: &'a mpsc::Sender<Response>,
+
+    no_delay: bool,
```

Comment:
Could you add a doc-string what this is doing?

Reply:
added docs about the semantics of the option in [16e2fe8](https://github.com/hashintel/hash/pull/4459/commits/16e2fe86ac3df4c076bd8b0ef4fa012c45dbb0b2)